src/utils/utils.test.tsx

```diff
@@ -0,0 +1,152 @@
+import featuredItemsData from "../data/featureditems.json";
+import {
+  generateRandomImageID,
+  getNumItems,
+  featuredImageID,
+  getAPIUri,
+  apiCall,
+} from "@/utils/utils";
+
+// test that this data returned in dataFromUri is valid:
+describe("featuredImageID Object", () => {
+  it("should return a defined imageID, uuid, title, and href if query param image id is not valid", async () => {
+    const imageID = featuredImageID("1234");
+    const apiUri = await getAPIUri("local_image_id", imageID);
```

Comment:
This means it's making real API requests in the test, right? these should be mocked.